library/std/src/sys/unix/thread_parker/netbsd.rs

```diff
@@ -0,0 +1,106 @@
+use crate::ffi::{c_int, c_void};
+use crate::pin::Pin;
+use crate::ptr::{null, null_mut};
+use crate::sync::atomic::{
+    AtomicU64,
+    Ordering::{Acquire, Relaxed, Release},
+};
+use crate::time::Duration;
+use libc::{_lwp_self, clockid_t, lwpid_t, time_t, timespec, CLOCK_MONOTONIC};
+
+extern "C" {
+    fn _lwp_park(
+        clock_id: clockid_t,
+        flags: c_int,
+        ts: *mut timespec,
+        unpark: lwpid_t,
+        hint: *const c_void,
+        unparkhint: *const c_void,
+    ) -> c_int;
+    fn _lwp_unpark(lwp: lwpid_t, hint: *const c_void) -> c_int;
+}
```

Comment:
It might be good to add these to [the `libc` crate](https://github.com/rust-lang/libc).

Reply:
The syscall behaviour can change slightly with newer versions (e.g. the `ts` argument is only modified on NetBSD 9.0 and above, while it is still the same symbol), so it might not necessarily be a good idea to advertise these functions in `libc`.

Reply:
I don't think that matters much for inclusion in the `libc` crate. It contains many symbols that aren't fully "stable" or only exist on certain versions of the operating system or underlying libc.

Anyway, this is not a blocker for this PR.